SyncExtension/UseCases/Folders/CreateFolderUseCase.swift

```diff
@@ -27,7 +28,12 @@ struct CreateFolderUseCase {
                 guard let parentFolderIdInt = Int(parentFolderId) else {
                     throw CreateItemError.NoParentIdFound
                 }
-                let createdFolder = try await driveAPI.createFolder(parentFolderId: parentFolderIdInt, folderName: itemTemplate.filename, debug: true)
+                
+                let filename = itemTemplate.filename as NSString
+                
+                let createdFolder = try await driveAPI.createFolder(parentFolderId: parentFolderIdInt, folderName: filename.deletingPathExtension, debug: true)
```

Comment:
Why the folder name comes from `deletetingPathExtension` ?

Reply:
Is just a secure way to strip an extension from the string in case it comes with one as folders should not have extension

Reply:
But if you are using that as a name, it means that you are using the entire path, which should not be the folder name, at least according to the first example of the provided link

Reply:
if you check where `filename` is coming from, is coming from `itemTemplate.filename` which is the actual filename of the item, what I want to prevent with this, is this kind of cases:

- A folder created with `folder.folder` will result in a folder created called `folder`
- A folder created with `/folder` (the OS will probably block this) will result in a folder created called `folder`

Is just a way to sanitize the filename for the folder,  doesn't mean the received value is a path like `/var/stuff/folder`, is just the NSString method name